src/rpc/core_rpc_server.cpp

```diff
@@ -2735,52 +2782,60 @@ namespace cryptonote
     for (auto &pubkey_info : pubkey_info_list)
     {
       COMMAND_RPC_GET_SERVICE_NODES::response::entry entry = {};
+      fill_sn_response_entry(entry, pubkey_info);
 
-      const auto proof = m_core.get_uptime_proof(pubkey_info.pubkey);
+      res.service_node_states.push_back(entry);
+    }
 
-      entry.service_node_pubkey           = string_tools::pod_to_hex(pubkey_info.pubkey);
-      entry.registration_height           = pubkey_info.info.registration_height;
-      entry.requested_unlock_height       = pubkey_info.info.requested_unlock_height;
-      entry.last_reward_block_height      = pubkey_info.info.last_reward_block_height;
-      entry.last_reward_transaction_index = pubkey_info.info.last_reward_transaction_index;
-      entry.last_uptime_proof             = proof.timestamp;
-      entry.service_node_version          = {proof.version_major, proof.version_minor, proof.version_patch};
-      entry.public_ip                     = string_tools::get_ip_string_from_int32(pubkey_info.info.public_ip);
-      entry.storage_port                  = pubkey_info.info.storage_port;
+    return true;
+  }
+  //------------------------------------------------------------------------------------------------------------------------------
+  bool core_rpc_server::on_get_n_service_nodes(const COMMAND_RPC_GET_N_SERVICE_NODES::request& req,
```

Comment:
Is there going to be a problem if you return service nodes that are still waiting for contribution when you request for N service nodes?

In particular for any of our internal services using this endpoint.

Reply:
Yes -- since the intention here is to use this for a seed node to get some active service nodes to connect to it should really only be including active ones.  (Or perhaps with a "include_inactive" flag that defaults to off -- though I'm not sure if there's a use case for getting some random, inactive SNs).

Reply:
Added `fully_funded_only` parameter to optionally exclude partially funded nodes.